Mathlib/Data/List/BigOperators/Basic.lean

```diff
@@ -143,6 +143,11 @@ theorem prod_map_mul {α : Type*} [CommMonoid α] {l : List ι} {f g : ι → α
 #align list.prod_map_mul List.prod_map_mul
 #align list.sum_map_add List.sum_map_add
 
+@[to_additive]
+theorem prod_map_one {α : Type*} [CommMonoid α] {l : List ι} :
+    (l.map fun _ => (1 : α)).prod = 1 := by
```

Comment:
This should be true for a `MulOneClass`

Reply:
Ok fair enough. The above assumptions all use `Monoid` so I have had to generalize there to fix this.

Reply:
Yep, `prod_cons` does not necessarily hold for `MulOneClass` so I need a new `prod_one_cons` to cover this.

Reply:
Can you move it to the `MulOneClass` section?